keps/sig-network/3766-referencegrant/README.md

```diff
@@ -0,0 +1,492 @@
+  - [ ] (R) Minimum Two Week Window for GA e2e tests to prove flake free
+- [ ] (R) Graduation criteria is in place
+  - [ ] (R) [all GA Endpoints](https://github.com/kubernetes/community/pull/1806) must be hit by [Conformance Tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/conformance-tests.md) 
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+[ReferenceGrant](https://gateway-api.sigs.k8s.io/api-types/referencegrant/) was
+developed by the [Gateway API subproject](https://gateway-api.sigs.k8s.io/) to
+enable certain object references to cross namespaces. More recently, it has also
+been [used by
+sig-storage](https://kubernetes.io/blog/2023/01/02/cross-namespace-data-sources-alpha/)
+to enable cross-namespace data sources.
+
+This KEP proposes moving ReferenceGrant from its current
+`gateway.networking.k8s.io` API group to a new `grants.authorization.k8s.io` API
+group.
+
+## Motivation
+
+Now that it's clear that ReferenceGrant is useful beyond just Gateway API, it
+would be good to formalize this model in a more neutral home, ideally sig-auth.
+At this point, each project that wants to enable cross-namespace references has
+to choose between introducing a dependency on Gateway API and creating a new
+resource that would largely duplicate ReferenceGrant. Both options would lead to
+confusion for Kubernetes users.
+
+### Goals
+
+* Move ReferenceGrant to sig-auth API Group
+* Clearly define how ReferenceGrant should be used, including both current use
+  cases and guidance for future use cases
+* Implement a library to ensure that ReferenceGrant is implemented consistently
+  by all controllers
+
+### Non-Goals
+
+* Add, remove, or change fields in the ReferenceGrant API
+* Develop an authorizer that will automatically implement ReferenceGrant for
+  all use cases
+
+## Proposal
+
+Move the existing ReferenceGrant resource into a new
+`grants.authorization.k8s.io` API group, defined within the Kubernetes code base
+as part of the 1.27 release.
+
+We will take this opportunity to clarify underspecified parts of the API, but
+will not add, change, or remove any fields as part of this transition.
+This resource will start with v1beta1 as the API version, matching the API version
+it already has within Gateway API.
```

Comment:
We didn't mention this inline, but, since Gateway API is a CRD using the `kubernetes.io` API group, ReferenceGrant has already had a Kubernetes API review pass to move from Alpha to Beta, that's why we're proposing it go straight to beta.

I can see that maybe sig-auth review is different, so maybe it does make sense to move this back to alpha. I'll update this part of the text to clarify that it's already gone through a general Kubernetes API review though.

Reply:
I think moving to a new SIG and becoming a general API dramatically changes the context of the API review itself.  Maybe I am being too cautious...

Reply:
I definitely have some feedback on this so I think Alpha is warranted. This type of thing can cause significant auth bugs.

Reply:
after reading it, I doubt this API survives without any changes :)